src/main/java/org/folio/calendar/service/CalendarService.java

```diff
@@ -0,0 +1,209 @@
+   * @return a {@link CalendarCollectionDTO CalendarCollectionDTO} with found calendars
+   */
+  public CalendarCollectionDTO getCalendarCollectionForServicePointsOrDateRange(
+    @CheckForNull List<UUID> servicePointIds,
+    LocalDate startDate,
+    LocalDate endDate,
+    Integer offset,
+    Integer limit
+  ) {
+    return calendarsToCalendarCollection(
+      calendarRepository.findWithServicePointsDateRangeAndPagination(
+        servicePointIds != null,
+        servicePointIds,
+        startDate,
+        endDate,
+        new CustomOffsetPageRequest(offset, limit)
+      ),
+      calendarRepository.countWithServicePointsDateRangeAndPagination(
+        servicePointIds != null,
+        servicePointIds,
+        startDate,
+        endDate
+      )
+    );
+  }
+
+  /**
+   * Get all the calendars based on a list of ids.  If not all calendars are
+   * found, a {@link DataNotFoundException DataNotFoundException} is thrown
+   *
+   * @param calendarIds a {@link java.util.List List} of calendars to search for
+   * @return a {@link CalendarCollectionDTO CalendarCollectionDTO} with found calendars
+   */
+  public CalendarCollectionDTO getCalendarCollectionForIdList(Set<UUID> calendarIds) {
+    List<Calendar> calendars = getCalendarsForIdList(calendarIds);
+    return calendarsToCalendarCollection(calendars, calendars.size());
+  }
+
+  /**
+   * Insert (or update) a calendar to the database
+   *
+   * @param calendar the calendar to insert/update/save
+   */
+  public void saveCalendar(Calendar calendar) {
+    this.calendarRepository.save(calendar);
+  }
+
+  /**
+   * Delete a calendar by its ID
+   *
+   * @param calendar the calendar to delete
+   */
+  public void deleteCalendar(Calendar calendar) {
+    this.calendarRepository.deleteCascadingById(calendar.getId());
+  }
+
+  /**
+   * Get objects for each date within a calendar's range representing opening
+   * or closure information
+   *
+   * @param servicePointId the service point ID to return opening information for
+   * @param startDate the first date to include
+   * @param endDate the last date to include
+   * @param includeClosed whether or not closed dates should be included
+   *        (exceptional closures are always included)
+   * @param offset pagination offset
+   * @param limit pagination limit
+   * @return a {@link SingleDayOpeningCollectionDTO SingleDayOpeningCollectionDTO}
+   * representing the dates' opening information
+   */
+  public SingleDayOpeningCollectionDTO getDailyOpeningCollection(
+    UUID servicePointId,
+    LocalDate startDate,
+    LocalDate endDate,
+    Boolean includeClosed,
+    Integer offset,
+    Integer limit
+  ) {
+    List<Calendar> relevantCalendars = calendarRepository.findWithServicePointsDateRangeAndPagination(
+      true,
+      Arrays.asList(servicePointId),
+      startDate,
+      endDate,
+      Pageable.unpaged()
+    );
+
+    // TreeMap makes sorting more efficient
```

Comment:
What sorting is performed on the tree map?

A brief review of the code suggests any sorting was done via streams, does that still utilise the capabilities of the map?